src/backend/renderer/gles/mod.rs

```diff
@@ -1407,6 +1395,20 @@ impl ExportMem for GlesRenderer {
         }
     }
 
+    fn can_read_texture(&mut self, texture: &Self::TextureId) -> Result<bool, GlesError> {
+        let old_target = self.target.take();
+
+        // if we can't bind the texture, we can't read it
+        let res = self.bind(texture.clone()).is_ok();
+
+        // restore
+        self.unbind()?;
+        self.target = old_target;
+        self.make_current()?;
```

Comment:
This is probably not an issue, but when unbind fails it won't restore the target, in case make_current fails it already restored it. I guess in the end it won't matter, but just noticed that difference.

Reply:
Yes, but unbind also sets the `target`. Which is why it is fine imo to just error out. This *shouldn't* happen and if it does it will almost certainly have messed up the state.

Reply:
Yeah, that was also my impression.